packages/camera/camera/lib/src/camera_controller.dart

```diff
@@ -294,11 +294,18 @@ class CameraController extends ValueNotifier<CameraValue> {
         'initialize was called on a disposed CameraController',
       );
     }
+
+    // dispose resources from previous camera description
+    if (_initCalled != null) {
+      await _initCalled;
+      _unawaited(CameraPlatform.instance.dispose(_cameraId));
```

Comment:
I don't like the idea that the `CameraController` calls `dispose` and is still remains usable. This doesn't seem like intended behavior. And now that I look at `setDescription`, is it possible to call `setDescription` instead of `initialize()` to initialize a camera? This also doesn't seem intended. 

I think it makes more sense to put this line in `setDescription` and not have this conditional.

Reply:
Ill change and test it with that under setDescription. 

I agree its weird that you could call setDescription to completely initialize the controller. Since setDescription is really just handling what you used to have to do manually when changing cameras -> re-initialize a controller with a new description. We could throw calls to setDescription if initialize has never been called?

Reply:
I agree. I think we should add the `_throwIfNotInitialized` method to the top of `setDescription`.

Reply:
Are you going to add `_throwIfNotInitialized` to the beginning of the method?